packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/FlipbookViewer/components/FlipbookControls.js

```diff
@@ -114,16 +116,32 @@ const FlipbookControls = ({
   }
 
   const resetTimeout = () => {
+    setIterationCounter(0)
     if (timeoutRef.current) {
       clearInterval(timeoutRef.current)
     }
   }
 
+  const handleIterationCounter = () => {
+    if (playIterations !== Infinity) {
+      setIterationCounter(iterationCounter + 1)
+    }
+  }
+
   useEffect(() => {
     if (playing) {
-      timeoutRef.current = setTimeout(() => {
-        handleNext()
-      }, 500 / playbackSpeed)
+      if (playIterations === Infinity) {
+        timeoutRef.current = setTimeout(() => {
+          handleNext()
+        }, 500 / playbackSpeed)
+      } else if (iterationCounter < playIterations * locations.length) {
```

Comment:
```suggestion
      if (iterationCounter < playIterations * locations.length) {
```

I think this change should be ok.